plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/button/frontend.tsx

```diff
@@ -146,23 +128,27 @@ const { state } = store< Store >( 'woocommerce/product-button', {
 				// After the cart is updated, sync the temporary number of items again.
 				context.temporaryNumberOfItems = state.numberOfItemsInTheCart;
 			} catch ( error ) {
-				const storeNoticeBlock =
-					document.querySelector( storeNoticeClass );
-
-				if ( ! storeNoticeBlock ) {
-					document
-						.querySelector( '.entry-content' )
-						?.prepend( createNoticeContainer() );
-				}
-
-				const domNode =
-					storeNoticeBlock ??
-					document.querySelector( storeNoticeClass );
-
 				const message = ( error as Error ).message;
+				const noticesStore = store< StoreNoticesStore >(
+					'woocommerce/store-notices'
+				);
 
-				if ( domNode ) {
-					injectNotice( domNode, decodeEntities( message ) );
+				// If the user deleted the hooked store notice block, the store won't be present
+				// and we should not add a notice.
+				if ( noticesStore ) {
+					// The old implementation always overwrites the last notice, so
+					// we remove the last notice before adding a new one.
+					if ( state.noticeId !== '' ) {
+						noticesStore.actions.removeNotice( state.noticeId );
+					}
```

Comment:
So, the assumption here is that we never want the errors produced by this block to be concatenated, right? They should always override the previous one.

Reply:
@luisherranz That's right, and I only do that to exactly match how it behaves right now.

Reply:
Ok, thanks.